ArrayIndexOutOfBoundsException in FrontendVersion if the minor version is not specified

Vaadin 15.0.0 application fails to start if there is a `@NpmPackage` dependency where the minor version is not specified, e.g.: `@NpmPackage(value = "@polymer/font-roboto", version="3")`

```
2020-03-11 19:08:26.775 ERROR 2524 --- [  restartedMain] o.a.c.c.C.[Tomcat].[localhost].[/]       : Exception sending context initialized event to listener instance of class [com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener]

java.lang.RuntimeException: Unable to initialize com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:166) ~[vaadin-spring-13.0.0.jar:na]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4683) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5146) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [na:na]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140) [na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [na:na]
	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140) [na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.apache.catalina.startup.Tomcat.start(Tomcat.java:459) [tomcat-embed-core-9.0.27.jar:9.0.27]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.initialize(TomcatWebServer.java:107) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.<init>(TomcatWebServer.java:88) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getTomcatWebServer(TomcatServletWebServerFactory.java:438) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getWebServer(TomcatServletWebServerFactory.java:191) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:180) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:153) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:544) [spring-context-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) [spring-boot-2.2.0.RELEASE.jar:2.2.0.RELEASE]
	at com.flowingcode.kms.Application.main(Application.java:14) [classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) [spring-boot-devtools-2.2.0.RELEASE.jar:2.2.0.RELEASE]
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
	at com.vaadin.flow.server.frontend.FrontendVersion.<init>(FrontendVersion.java:127) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.frontend.NodeUpdater.toVersion(NodeUpdater.java:364) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.frontend.NodeUpdater.handleExistingVaadinDep(NodeUpdater.java:335) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.frontend.NodeUpdater.addDependency(NodeUpdater.java:318) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.updatePackageJsonDependencies(TaskUpdatePackages.java:115) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.execute(TaskUpdatePackages.java:94) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:572) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.server.startup.DevModeInitializer.initDevModeHandler(DevModeInitializer.java:331) ~[flow-server-3.0.0.jar:3.0.0]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$DevModeServletContextListener.failFastContextInitialized(VaadinServletContextInitializer.java:345) ~[vaadin-spring-13.0.0.jar:na]
	at com.vaadin.flow.spring.VaadinServletContextInitializer$FailFastServletContextListener.contextInitialized(VaadinServletContextInitializer.java:162) ~[vaadin-spring-13.0.0.jar:na]
	... 43 common frames omitted
```

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendVersion.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendVersion.java
@@ -58,9 +58,9 @@ public class FrontendVersion
      * Create a version of format "major.minor.0".
      *
      * @param major
-     *         major version
+     *            major version
      * @param minor
-     *         minor version
+     *            minor version
      */
     public FrontendVersion(int major, int minor) {
         this(major, minor, 0);
@@ -70,11 +70,11 @@ public class FrontendVersion
      * Create a version of format "major.minor.revision".
      *
      * @param major
-     *         major version
+     *            major version
      * @param minor
-     *         minor version
+     *            minor version
      * @param revision
-     *         revision number
+     *            revision number
      */
     public FrontendVersion(int major, int minor, int revision) {
         this(major, minor, revision, "");
@@ -84,16 +84,16 @@ public class FrontendVersion
      * Create a version of format "major.minor.revision.build"
      *
      * @param major
-     *         major version
+     *            major version
      * @param minor
-     *         minor version
+     *            minor version
      * @param revision
-     *         revision number
+     *            revision number
      * @param build
-     *         build identifier
+     *            build identifier
      */
     public FrontendVersion(int major, int minor, int revision, String build) {
-        if(build.isEmpty()) {
+        if (build.isEmpty()) {
             this.version = major + "." + minor + "." + revision;
         } else {
             this.version = major + "." + minor + "." + revision + "." + build;
@@ -111,10 +111,13 @@ public class FrontendVersion
      * Versions are normalized and any caret or tildes will not be considered.
      *
      * @param version
-     *         version string as "major.minor.revision[.build]"
+     *            version string as "major.minor.revision[.build]"
      */
     public FrontendVersion(String version) {
         Objects.requireNonNull(version);
+        if (version.isEmpty()) {
+            throw new NumberFormatException(getInvalidVersionMessage(version));
+        }
         if (!Character.isDigit(version.charAt(0))) {
             this.version = version.substring(1).trim();
         } else {
@@ -124,16 +127,24 @@ public class FrontendVersion
         final String[] digits = this.version.split("[-.]", 4);
         try {
             majorVersion = Integer.parseInt(digits[0]);
-            minorVersion = Integer.parseInt(digits[1]);
         } catch (NumberFormatException nfe) {
-            throw new NumberFormatException(
-                    String.format("'%s' is not a valid version!", version));
+            throw new NumberFormatException(getInvalidVersionMessage(version));
+        }
+        if (digits.length >= 2) {
+            try {
+                minorVersion = Integer.parseInt(digits[1]);
+            } catch (NumberFormatException nfe) {
+                throw new NumberFormatException(
+                        getInvalidVersionMessage(version));
+            }
+        } else {
+            minorVersion = 0;
         }
         int revisionNumber;
         String build = "";
         try {
-            revisionNumber =
-                    digits.length >= 3 ? Integer.parseInt(digits[2]) : 0;
+            revisionNumber = digits.length >= 3 ? Integer.parseInt(digits[2])
+                    : 0;
             if (digits.length == 4) {
                 build = digits[3];
             }
@@ -196,11 +207,11 @@ public class FrontendVersion
     }
 
     /**
-     * Check if this version is older than given version.
-     * Will return false if equals or is newer.
+     * Check if this version is older than given version. Will return false if
+     * equals or is newer.
      *
      * @param otherVersion
-     *         version to check against
+     *            version to check against
      * @return true if this is older than otherVersion
      */
     public boolean isOlderThan(FrontendVersion otherVersion) {
@@ -208,11 +219,11 @@ public class FrontendVersion
     }
 
     /**
-     * Check if this version is newer than given version.
-     * Will return false if equals or is older.
+     * Check if this version is newer than given version. Will return false if
+     * equals or is older.
      *
      * @param otherVersion
-     *         version to check against
+     *            version to check against
      * @return true if this is newer than otherVersion
      */
     public boolean isNewerThan(FrontendVersion otherVersion) {
@@ -223,7 +234,7 @@ public class FrontendVersion
      * Check if this and the given version are equal to each other.
      *
      * @param otherVersion
-     *         version to test equals with
+     *            version to test equals with
      * @return true if parsed version parts are exactly the same
      */
     public boolean isEqualTo(FrontendVersion otherVersion) {
@@ -236,8 +247,8 @@ public class FrontendVersion
             FrontendVersion other = (FrontendVersion) obj;
             return majorVersion == other.getMajorVersion()
                     && minorVersion == other.getMinorVersion()
-                    && revision == other.getRevision() && buildIdentifier
-                    .equals(other.getBuildIdentifier());
+                    && revision == other.getRevision()
+                    && buildIdentifier.equals(other.getBuildIdentifier());
 
         }
         return false;
@@ -250,11 +261,11 @@ public class FrontendVersion
     }
 
     /**
-     * Compare version numbers and return order as -1, 0 and 1.
-     * Where this version is older, equals, newer than given version.
+     * Compare version numbers and return order as -1, 0 and 1. Where this
+     * version is older, equals, newer than given version.
      *
      * @param other
-     *         version to compare against this version
+     *            version to compare against this version
      * @return -1 this is older, 0 versions equal, 1 this is newer
      */
     @Override
@@ -271,12 +282,17 @@ public class FrontendVersion
         if (this.buildIdentifier != other.buildIdentifier) {
             if (buildIdentifier.isEmpty() && !other.buildIdentifier.isEmpty()) {
                 return 1;
-            } else if (!buildIdentifier.isEmpty() && other.buildIdentifier
-                    .isEmpty()) {
+            } else if (!buildIdentifier.isEmpty()
+                    && other.buildIdentifier.isEmpty()) {
                 return -1;
             }
             return buildIdentifier.compareToIgnoreCase(other.buildIdentifier);
         }
         return 0;
     }
+
+    private String getInvalidVersionMessage(String version) {
+        return String.format("'%s' is not a valid version!", version);
+    }
+
 }

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendVersionTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendVersionTest.java
@@ -47,8 +47,7 @@ public class FrontendVersionTest {
 
         fromString = new FrontendVersion("1.1.alpha12");
         fromConstructor = new FrontendVersion(1, 1, 0, "alpha12");
-        assertTrue(
-                "Major-Minor version with build identifier didn't match",
+        assertTrue("Major-Minor version with build identifier didn't match",
                 fromConstructor.equals(fromString));
 
         fromString = new FrontendVersion("12.3.5.alpha12");
@@ -69,8 +68,7 @@ public class FrontendVersionTest {
 
         fromString = new FrontendVersion("1.1.alpha12");
         fromConstructor = new FrontendVersion(1, 1, 0, "alpha12");
-        assertTrue(
-                "Major-Minor version with build identifier didn't match",
+        assertTrue("Major-Minor version with build identifier didn't match",
                 fromConstructor.isEqualTo(fromString));
 
         fromString = new FrontendVersion("12.3.5.alpha12");
@@ -84,6 +82,21 @@ public class FrontendVersionTest {
         new FrontendVersion("12.0b.1");
     }
 
+    @Test(expected = NumberFormatException.class)
+    public void notANumber_throwsException() {
+        new FrontendVersion("a");
+    }
+
+    @Test(expected = NumberFormatException.class)
+    public void emptyString_throwsException() {
+        new FrontendVersion("");
+    }
+
+    @Test
+    public void onlyMajorVersion_allVersoinNumbersAreCalculated() {
+        assertVersion(new FrontendVersion("3"), 3, 0, 0, "");
+    }
+
     @Test
     public void versionHandlesTildeAndCaretVersions() {
         assertVersion(new FrontendVersion("~1.3.0-beta1"), 1, 3, 0, "beta1");
@@ -95,31 +108,47 @@ public class FrontendVersionTest {
     public void olderVersionIsCalculatedCorrectly() {
         FrontendVersion test = new FrontendVersion("2.2.0");
 
-        assertTrue("Should be older due to revision", test.isOlderThan(new FrontendVersion("2.2.1")));
-        assertTrue("Should be older due to minor",test.isOlderThan(new FrontendVersion("2.3.0")));
-        assertTrue("Should be older due to major", test.isOlderThan(new FrontendVersion("3.2.0")));
+        assertTrue("Should be older due to revision",
+                test.isOlderThan(new FrontendVersion("2.2.1")));
+        assertTrue("Should be older due to minor",
+                test.isOlderThan(new FrontendVersion("2.3.0")));
+        assertTrue("Should be older due to major",
+                test.isOlderThan(new FrontendVersion("3.2.0")));
 
-        assertFalse("Should be newer as target has buildIdentifier", test.isOlderThan(new FrontendVersion("2.2.0-alpha1")));
+        assertFalse("Should be newer as target has buildIdentifier",
+                test.isOlderThan(new FrontendVersion("2.2.0-alpha1")));
 
-        assertFalse("Should be newer due to major", test.isOlderThan(new FrontendVersion("1.2.0")));
-        assertFalse("Should be newer due to minor", test.isOlderThan(new FrontendVersion("2.1.0")));
-        assertFalse("Should be newer due to minor", test.isOlderThan(new FrontendVersion("2.1.2")));
-        assertFalse("Should be newer due to minor", test.isOlderThan(new FrontendVersion("1.5.2")));
+        assertFalse("Should be newer due to major",
+                test.isOlderThan(new FrontendVersion("1.2.0")));
+        assertFalse("Should be newer due to minor",
+                test.isOlderThan(new FrontendVersion("2.1.0")));
+        assertFalse("Should be newer due to minor",
+                test.isOlderThan(new FrontendVersion("2.1.2")));
+        assertFalse("Should be newer due to minor",
+                test.isOlderThan(new FrontendVersion("1.5.2")));
 
-        assertFalse("Should be newer by major even with buildIdentifier", test.isOlderThan(new FrontendVersion("1.2.0-alpha1")));
+        assertFalse("Should be newer by major even with buildIdentifier",
+                test.isOlderThan(new FrontendVersion("1.2.0-alpha1")));
     }
+
     @Test
     public void newerVersionIsCalculatedCorrectly() {
         FrontendVersion test = new FrontendVersion("2.2.2");
 
-        assertTrue("Should be newer due to revision", test.isNewerThan(new FrontendVersion("2.2.1")));
-        assertTrue("Should be newer due to minor", test.isNewerThan(new FrontendVersion("2.1.2")));
-        assertTrue("Should be newer due to major", test.isNewerThan(new FrontendVersion("1.2.2")));
+        assertTrue("Should be newer due to revision",
+                test.isNewerThan(new FrontendVersion("2.2.1")));
+        assertTrue("Should be newer due to minor",
+                test.isNewerThan(new FrontendVersion("2.1.2")));
+        assertTrue("Should be newer due to major",
+                test.isNewerThan(new FrontendVersion("1.2.2")));
 
-        assertTrue("Should be newer as target has buildIdentifier", test.isNewerThan(new FrontendVersion("2.2.2-alpha1")));
+        assertTrue("Should be newer as target has buildIdentifier",
+                test.isNewerThan(new FrontendVersion("2.2.2-alpha1")));
 
-        assertFalse("Should be older due to major", test.isNewerThan(new FrontendVersion("3.2.2")));
-        assertFalse("Should be older due to minor", test.isNewerThan(new FrontendVersion("2.3.2")));
+        assertFalse("Should be older due to major",
+                test.isNewerThan(new FrontendVersion("3.2.2")));
+        assertFalse("Should be older due to minor",
+                test.isNewerThan(new FrontendVersion("2.3.2")));
     }
 
     @Test
@@ -127,16 +156,23 @@ public class FrontendVersionTest {
 
         FrontendVersion test = new FrontendVersion("2.0.0-RC1");
 
-        assertTrue("2.0.0 should be newer that RC1",test.isOlderThan(new FrontendVersion("2.0.0")));
-        assertTrue("RC2 should be newer that RC1",test.isOlderThan(new FrontendVersion("2.0.0-RC2")));
-        assertFalse("beta5 should be older than RC1",test.isOlderThan(new FrontendVersion("2.0.0-beta5")));
-        assertFalse("alpha4 should be older than RC1", test.isOlderThan(new FrontendVersion("2.0.0-alpha4")));
+        assertTrue("2.0.0 should be newer that RC1",
+                test.isOlderThan(new FrontendVersion("2.0.0")));
+        assertTrue("RC2 should be newer that RC1",
+                test.isOlderThan(new FrontendVersion("2.0.0-RC2")));
+        assertFalse("beta5 should be older than RC1",
+                test.isOlderThan(new FrontendVersion("2.0.0-beta5")));
+        assertFalse("alpha4 should be older than RC1",
+                test.isOlderThan(new FrontendVersion("2.0.0-alpha4")));
 
         test = new FrontendVersion("2.0.0");
 
-        assertFalse("RC2 should be older that 2.0.0",test.isOlderThan(new FrontendVersion("2.0.0-RC2")));
-        assertFalse("beta5 should be older than 2.0.0",test.isOlderThan(new FrontendVersion("2.0.0-beta5")));
-        assertFalse("alpha4 should be older than 2.0.0", test.isOlderThan(new FrontendVersion("2.0.0-alpha4")));
+        assertFalse("RC2 should be older that 2.0.0",
+                test.isOlderThan(new FrontendVersion("2.0.0-RC2")));
+        assertFalse("beta5 should be older than 2.0.0",
+                test.isOlderThan(new FrontendVersion("2.0.0-beta5")));
+        assertFalse("alpha4 should be older than 2.0.0",
+                test.isOlderThan(new FrontendVersion("2.0.0-alpha4")));
     }
 
     @Test
@@ -144,16 +180,23 @@ public class FrontendVersionTest {
 
         FrontendVersion test = new FrontendVersion("2.0.0-alpha2");
 
-        assertTrue("alpha2 should be newer than alpha1", test.isNewerThan(new FrontendVersion("2.0.0-alpha1")));
-        assertFalse("alpha2 should be older than 2.0.0", test.isNewerThan(new FrontendVersion("2.0.0")));
-        assertFalse("alpha2 should be older than beta1",test.isNewerThan(new FrontendVersion("2.0.0-beta1")));
-        assertFalse("alpha2 should be older than RC1",test.isNewerThan(new FrontendVersion("2.0.0-RC1")));
+        assertTrue("alpha2 should be newer than alpha1",
+                test.isNewerThan(new FrontendVersion("2.0.0-alpha1")));
+        assertFalse("alpha2 should be older than 2.0.0",
+                test.isNewerThan(new FrontendVersion("2.0.0")));
+        assertFalse("alpha2 should be older than beta1",
+                test.isNewerThan(new FrontendVersion("2.0.0-beta1")));
+        assertFalse("alpha2 should be older than RC1",
+                test.isNewerThan(new FrontendVersion("2.0.0-RC1")));
 
         test = new FrontendVersion("2.0.0");
 
-        assertTrue("2.0.0 should be newer than alpha1", test.isNewerThan(new FrontendVersion("2.0.0-alpha1")));
-        assertTrue("2.0.0 should be older than beta1",test.isNewerThan(new FrontendVersion("2.0.0-beta1")));
-        assertTrue("2.0.0 should be older than RC1",test.isNewerThan(new FrontendVersion("2.0.0-RC1")));
+        assertTrue("2.0.0 should be newer than alpha1",
+                test.isNewerThan(new FrontendVersion("2.0.0-alpha1")));
+        assertTrue("2.0.0 should be older than beta1",
+                test.isNewerThan(new FrontendVersion("2.0.0-beta1")));
+        assertTrue("2.0.0 should be older than RC1",
+                test.isNewerThan(new FrontendVersion("2.0.0-RC1")));
     }
 
     private void assertVersion(FrontendVersion version, int major, int minor,

